lib/core/connector_settings.rb

```diff
@@ -114,6 +114,16 @@ def valid_index_name?
       index_name&.start_with?(Utility::Constants::CONTENT_INDEX_PREFIX)
     end
 
+    def ready_for_sync?
+      Connectors::REGISTRY.registered?(service_type) &&
```

Comment:
I disagree because with a new setup anyone can index a job document into the sync job index we need to make sure that the connector is ready to pick up the job.